Library/Homebrew/formula_auditor.rb

```diff
@@ -243,8 +243,13 @@ def audit_license
         user, repo = get_repo_data(%r{https?://github\.com/([^/]+)/([^/]+)/?.*})
         return if user.blank?
 
-        tag = SharedAudits.github_tag_from_url(formula.stable.url)
+        tag = if formula.stable.url.start_with?("https://files.pythonhosted.org/")
+          formula.stable.url.match(%r{/([^/]*)\.tar\.gz$})[1]
```

Comment:
This won't work for linked PR (`codelimit`) as it will result in `tag = "codelimit-0.8.1"` and just skip the check, which probably isn't what we want as default behavior.
```console
❯ brew ruby -e 'require "formula"; p Formula["codelimit"].stable.url.match(%r{/([^/]*)\.tar\.gz$})[1]'
"codelimit-0.8.1"
```

If we are skipping license check, it should be explicit (i.e. audit exception) in order for maintainers to manually confirm.

`codelimit` would need to be run on `tag = "v0.8.1"`, which isn't easy to get from the PyPI tarball (i.e. would need to try multiple attempts perhaps based on "common" formats of version).
```console
❯ brew ruby -e '["codelimit-0.8.1", "0.8.1", "v0.8.1", nil].each { |tag| p GitHub.get_repo_license("getcodelimit", "codelimit", ref: tag) }'
nil
nil
"ISC"
"GPL-3.0"
```

Anyway, when we are guessing `tag` from non-GitHub URL, we should always fallback on `tag = nil` if `GitHub.get_repo_license` returns `nil` to avoid unwanted bypass.

---

On side note, if upstream was willing to upload license to PyPI, we could consider adding a check there and prioritize it over GitHub, but may need to transform info to match our expected SPDX format and would require sufficiently standardized output.
```console
❯ curl -sL "https://pypi.org/pypi/codelimit/0.8.1/json" | jq '.info.license'
""

❯ curl -sL "https://pypi.org/pypi/awscli/json" | jq '.info.license'
"Apache License 2.0"
```